include/dlaf/permutations/general/impl.h

```diff
@@ -288,7 +289,7 @@ void all2allData(common::Pipeline<comm::Communicator>& sub_task_chain, int nrank
         pika::this_thread::experimental::sync_wait(ex::when_all_vector(std::move(all_comms)));
       };
 
-  ex::when_all(sub_task_chain(), std::forward<SendCountsSender>(send_counts_sender),
+  ex::when_all(sub_task_chain.readwrite(), std::forward<SendCountsSender>(send_counts_sender),
```

Comment:
This implementation is currently exploiting a `pipeline.readwrite()` call to mimic what now can be done with a `sub_pipeline()`.

Indeed, currently we pass a sender of a Communicator, and within the task we "manually" reuse it, as if it was a sequence of `read()` calls on a `sub_pipeline()`.

Reply:
I had a quick look at this and indeed it seems like it was an easy change. At least the permutations tests pass for me locally. This was changed in https://github.com/eth-cscs/DLA-Future/pull/993/commits/b11bbc37a9a88ecebd9764df21f08d22ce0076c8.